nemo/core/classes/modelPT.py

```diff
@@ -240,6 +241,22 @@ def register_artifact(
 
         return self._save_restore_connector.register_artifact(self, config_path, src, verify_src_exists)
 
+    def has_artifacts(self) -> bool:
+        """Returns True if model has artifacts registered"""
+        return hasattr(self, 'artifacts') and self.artifacts is not None and len(self.artifacts) > 0
```

Comment:
What is the reason to check everywhere `hasattr` + `not None` for `artifacts`? Can we just assign `self.artifacts = {}` in `ModelPT` constructor?

Reply:
Im ok with this, but instead of just setting it, inside modelpt do a hasattr check and only then set to None if it doesn't have the attr. Actually, it might cause issues but we can think about it (maybe another PR)

There is a case where we need to register before calling the constructor (look at ASRBPEMixin)